core/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -787,7 +797,21 @@ public void writeTo(StreamOutput out) throws IOException {
             nodes.writeTo(out);
             metaData.writeTo(out);
             blocks.writeTo(out);
-            customs.writeTo(out);
+            Diff<ImmutableOpenMap<String, Custom>> customsDiff = customs;
+            if (out.getVersion().before(SnapshotDeletionsInProgress.VERSION_INTRODUCED)) {
+                customsDiff = removeSnapshotDeletionsCustomDiff(customsDiff);
```

Comment:
I think this should be handled in a generic fashion on the custom's serialization level, but it might be better to do in a follow-up PR after I merge #22336 in.